### PR TITLE
Minor cleanup to PairValueIndex and attach_indices

### PR DESCRIPTION
**Pull request by aprokop**
- Do not ignore the second template arg of PairValueIndex in LegacyCallbackWrapper
- Match the default for `Index` in `attach_indices` to the default in `PairValueIndex`

### CODE REVIEW

File: src/details/ArborX_PairValueIndex.hpp [hunk 1]
```diff
@@ -56,7 +56,9 @@ class AttachIndices
   auto size() const { return _data.size(); }
 };
 
-template <typename Values, typename Index = unsigned>
+// Make sure the default Index matches the default in PairValueIndex
```
Comment (dalg24) [inline src/details/ArborX_PairValueIndex.hpp:59]: I don't understand what you mean by make sure?
Should that be a static assertion?

File: src/details/ArborX_DetailsLegacy.hpp [hunk 2]
```diff
@@ -61,27 +61,29 @@ struct LegacyCallbackWrapper
...
+             PairValueIndex<Value, Index> const &value) const
   {
     return _callback(predicate, value.index);
   }
 
-  template <typename Predicate, typename Value, typename Output>
-  KOKKOS_FUNCTION auto operator()(Predicate const &predicate,
-                                  PairValueIndex<Value> const &value,
+  template <typename Predicate, typename Value, typename Index, typename Output>
+  KOKKOS_FUNCTION void operator()(Predicate const &predicate,
```
Comment (dalg24) [inline src/details/ArborX_DetailsLegacy.hpp:73]: Why did you change the return type?

File: src/details/ArborX_PairValueIndex.hpp (hunk 1 above)
Comment (aprokop) [reply src/details/ArborX_PairValueIndex.hpp:59]: They must actually match when constructing hierarchy. However, we cannot really enforce it, imho, as user is responsible for specifying both. The next best thing is to at least make sure that the default values for both match.

File: src/details/ArborX_DetailsLegacy.hpp (hunk 2 above)
Comment (aprokop) [reply src/details/ArborX_DetailsLegacy.hpp:73]: For clarity. Our 3-arg callback is only allowed to return void`.

Comment (dalg24) [reply src/details/ArborX_DetailsLegacy.hpp:73]: Do you remember why we required it to return `void`?

File: src/details/ArborX_PairValueIndex.hpp (hunk 1 above)
Comment (dalg24) [reply src/details/ArborX_PairValueIndex.hpp:59]: Is this about making sure that the specified default (`unsigned`) matches the default from `PairValueIndex`?

Comment (aprokop) [reply src/details/ArborX_PairValueIndex.hpp:59]: Yes.

File: src/details/ArborX_DetailsLegacy.hpp (hunk 2 above)
Comment (aprokop) [reply src/details/ArborX_DetailsLegacy.hpp:73]: Because the semantics of early termination when doing two pass are unclear.

**Review by dalg24 — APPROVED** [review]
